docs-src/dev-guide/sdk-features.md

```diff
@@ -0,0 +1,571 @@
+| --------------------------------- | --- | ---- | ---------- | ------ | ---- | --- |
+| [Health service](#health-service) | Yes | Yes  | Yes        | Yes    | Yes  | No  |
+
+A health service in the context of Temporal is a component responsible for checking the health of the Frontend Service.
+It ensures the proper functioning of the Cluster by returning a list of cluster metrics.
+The health checks for the Temporal Cluster can be set up using TCP or gRPC on port 7233.
+The Matching Service is responsible for hosting user-facing Task Queues, while the History Service persists Workflow Execution state.
+The Worker service performs background processing for replication queue and system Workflows [2].
+
+## Heartbeats
+
+| Feature                   | Go  | Java | TypeScript | Python | .Net | PHP |
+| ------------------------- | --- | ---- | ---------- | ------ | ---- | --- |
+| [Heartbeats](#heartbeats) | Yes | Yes  | Yes        | Yes    | Yes  | Yes |
+
+A Heartbeat is a mechanism used in the Temporal Platform to ensure the progress and timely execution of Activities.
+It involves periodic pings from the Activity Worker to the Temporal Cluster.
+The Heartbeat Timeout specifies the maximum time between Activity Heartbeats.
+Heartbeats can be used to store progress information or any other serializable object.
+By Heartbeating within a specified interval lower than the Heartbeat Timeout, Activities can execute for a long time and fail fast.
+Heartbeats help prevent Activity Timeouts and enable efficient retries.
+
+## Interceptors
+
+| Feature                       | Go  | Java | TypeScript | Python | .Net | PHP |
+| ----------------------------- | --- | ---- | ---------- | ------ | ---- | --- |
+| [Interceptors](#interceptors) | Yes | Yes  | Yes        | Yes    | Yes  | No  |
+
+Interceptors are a mechanism for modifying inbound and outbound SDK calls.
+Interceptors are commonly used to add tracing and authorization to the scheduling and execution of Workflows and Activities.
+You can compare these to "middleware" in other frameworks.
+
+The TypeScript SDK comes with an optional interceptor package that adds tracing with [OpenTelemetry](https://www.npmjs.com/package/@temporalio/interceptors-opentelemetry).
+See how to use it in the [interceptors-opentelemetry](https://github.com/temporalio/samples-typescript/tree/main/interceptors-opentelemetry) code sample.
+
+The [Go SDK](https://github.com/temporalio/sdk-go) provides support for distributed tracing with **_Interceptors_**.
+Interceptors uses Temporal headers to create a call graph of a [Workflow](/workflows#), along with its [Activities](/activities#) and [Child Workflows](/workflows#child-workflow).
+
+There are several tracing implementations supported by the Temporal Go SDK.
+For an [OpenTracing](https://pkg.go.dev/go.temporal.io/sdk/contrib/opentracing) Interceptor, use `opentracing.NewInterceptor(opentracing.TracerOptions{})` to create a `TracingInterceptor`.
+
+## Message passing
```

Comment:
I don't think we like to refer to those features as "message passing". Regardless, they will be in all SDKs, so is there any value in putting here?